components/filter/filter.js

```diff
@@ -0,0 +1,270 @@
+		this.addEventListener('d2l-hierarchical-view-hide-start', this._stopPropagation);
+		this.addEventListener('d2l-hierarchical-view-show-complete', this._stopPropagation);
+		this.addEventListener('d2l-hierarchical-view-show-start', this._stopPropagation);
+		this.addEventListener('d2l-hierarchical-view-resize', this._stopPropagation);
+	}
+
+	render() {
+		const header = this._buildHeader();
+		const dimensions = this._buildDimensions();
+
+		return html`
+			<d2l-dropdown-button-subtle
+				@d2l-dropdown-close="${this._handleDropdownClose}"
+				@d2l-dropdown-open="${this._stopPropagation}"
+				@d2l-dropdown-position="${this._stopPropagation}"
+				text="${this.localize('components.filter.filters')}"
+				?disabled="${this.disabled}">
+				<d2l-dropdown-menu min-width="285" max-width="420" no-padding-header>
+					${header}
+					<d2l-menu label="${this.localize('components.filter.filters')}">
+						<slot
+							@d2l-filter-dimension-data-change="${this._handleDimensionDataChange}"
+							@d2l-filter-dimension-slot-change="${this._handleSlotChange}"
+							@slotchange="${this._handleSlotChange}"
+						></slot>
+						${dimensions}
+					</d2l-menu>
+				</d2l-dropdown-menu>
+			</d2l-dropdown-button-subtle>
+		`;
+	}
+
+	_buildDimension(dimension) {
+		let dimensionHTML;
+		switch (dimension.type) {
+			case 'd2l-filter-dimension-set':
+				dimensionHTML = this._createSetDimension(dimension);
+				break;
+		}
+		return html`
+			<d2l-hierarchical-view
+				@d2l-hierarchical-view-show-complete="${this._handleDimensionShowComplete}"
+				@d2l-hierarchical-view-show-start="${this._handleDimensionShowStart}"
+				id="${dimension.key}">
+
+				${dimensionHTML}
+			</d2l-hierarchical-view>
+		`;
+	}
+
+	_buildDimensions() {
+		return this._dimensions.map((dimension) => {
+			const builtDimension = this._buildDimension(dimension);
+			return html`<d2l-menu-item text="${dimension.text}">
+				${builtDimension}
+			</d2l-menu-item>`;
+		});
+	}
+
+	_buildHeader() {
+		if (!this._activeDimensionKey) return null;
+
+		const dimensionText = this._dimensions.find(dimension => dimension.key === this._activeDimensionKey).text;
+		return html`
+			<div slot="header">
+				<div class="d2l-filter-dimension-header">
+					<d2l-button-icon
+						@click="${this._handleDimensionHide}"
+						icon="tier1:chevron-left"
+						text="${this.localize('components.menu-item-return.returnCurrentlyShowing', 'menuTitle', dimensionText)}">
+					</d2l-button-icon>
+					<div class="d2l-filter-dimension-header-text d2l-body-standard">${dimensionText}</div>
+				</div>
+			</div>
+		`;
+	}
+
+	_createSetDimension(dimension) {
+		return html`
+			<d2l-list
+				@d2l-list-selection-change="${this._handleChangeSetDimension}"
+				extend-separators>
+
+				${dimension.values.map(item => html`
+					<d2l-list-item
+						key="${item.key}"
+						selectable
+						?selected="${item.selected}"
+						slim>
+
```

Comment:
Does this whitespace get rendered?  I am assuming it gets ignored... 🤔

Reply:
I was going to mention this as a "nit" too... I'm always paranoid about whitespace in HTML templates as they can result in extra text nodes, but likely not a real issue.

Reply:
I can remove it lol, I just hate the look of having the indented properties and then a new component at the same level.  It means I can't tell the hierarchy _quite_ as quick. This would solve that, but I don't think we do this anywhere and it's also not very nice looking:
```
	<d2l-list-item
		key="${item.key}"
		selectable
		?selected="${item.selected}"
		slim
	>
		<div class="d2l-filter-dimension-set-value-text">${item.text}</div>
	</d2l-list-item>
```